serializers/src/glTF/2.0/Extensions/KHR_lights_punctual.ts

```diff
@@ -0,0 +1,185 @@
+                    }
+                    if (babylonLight.falloffType != Light.FALLOFF_GLTF) {
+                        Tools.Warn(`${context}: Light falloff for ${babylonLight.name} does not match the ${NAME} specification!`);
+                    }
+                    light = {
+                        type: lightType
+                    };
+                    if (!babylonLight.diffuse.equalsFloats(1.0, 1.0, 1.0)) {
+                        light.color = babylonLight.diffuse.asArray();
+                    }
+                    if (!(babylonLight.intensity == 1.0)) {
+                        light.intensity = babylonLight.intensity;
+                    }
+                    if (!(babylonLight.range == Number.MAX_VALUE)) {
+                        light.range = babylonLight.range;
+                    }
+
+                    if (lightType === LightType.SPOT) {
+                        let babylonSpotLight = babylonLight as SpotLight;
+                        if (!(babylonSpotLight.angle == Math.PI / 2.0)) {
+                            if (light.spot == null) {
+                                light.spot = {};
+                            }
+                            light.spot.outerConeAngle = babylonSpotLight.angle / 4;
+                        }
+                        if (!(babylonSpotLight.innerAngle == 0)) {
+                            if (light.spot == null) {
+                                light.spot = {};
+                            }
+                            light.spot.innerConeAngle = babylonSpotLight.innerAngle / 2;
+                        }
+                        let rotation = babylonSpotLight.getRotation().clone();
+                        if (!rotation.equals(Vector3.Zero())) {
+                            if (this._exporter._convertToRightHandedSystem) {
+                                rotation.z *= -1;
+                            }
+                            node.rotation = Quaternion.FromEulerAngles(rotation.x, rotation.y, rotation.z).asArray();
+                        }
+                    }
+                    else if (lightType === LightType.DIRECTIONAL) {
+                        let babylonDirectionalLight = babylonLight as DirectionalLight;
+                        let rotation = babylonDirectionalLight.getRotation().clone();
+                        if (!rotation.equals(Vector3.Zero())) {
+                            if (this._exporter._convertToRightHandedSystem) {
+                                rotation.z *= -1;
+                            }
+                            node.rotation = Quaternion.FromEulerAngles(rotation.x, rotation.y, rotation.z).asArray();
+                        }
+                    }
+                    if (this._lights == null) {
+                        this._lights = {
+                            lights: []
+                        };
+                    }
+
+                    this._lights.lights.push(light);

```

Comment:
can lights be shared or instanced?

Reply:
@bghgary good point.  I can check if the light has already been serialized first.

Reply:
Ideally, I think we should check if the lights have the same properties and don't export a light with the same properties.

Reply:
This looks to be a bit complex to implement, but maybe an optimization for a future update.